packages/notifications/push/amplify_push_notifications/lib/amplify_push_notifications.dart

```diff
@@ -0,0 +1,227 @@
+          .createChild('AmplifyPushNotification');
+  final ServiceProviderClient _serviceProviderClient;
+
+  late final Stream<String> _onTokenReceived;
+  late final Stream<PushNotificationMessage> _onForegroundNotificationReceived;
+  late final Stream<PushNotificationMessage> _onBackgroundNotificationReceived;
+  late final Stream<PushNotificationMessage> _onNotificationOpened;
+  late final Stream<PushNotificationMessage> _onLaunchNotificationOpened;
+
+  bool _isConfigured = false;
+
+  @override
+  Stream<String> get onTokenReceived => _onTokenReceived;
+
+  @override
+  Stream<PushNotificationMessage> get onNotificationReceivedInForeground =>
+      _onForegroundNotificationReceived;
+
+  @override
+  Stream<PushNotificationMessage> get onNotificationOpened =>
+      _onNotificationOpened;
+
+  @override
+  void onNotificationReceivedInBackground(OnRemoteMessageCallback callback) {
+    throw UnimplementedError(
+      'onNotificationReceivedInBackground() has not been implemented.',
+    );
+  }
+
+  @override
+  Future<void> configure({
+    AmplifyConfig? config,
+    required AmplifyAuthProviderRepository authProviderRepo,
+  }) async {
+    // Parse config values from amplifyconfiguration.json
+    final analyticsConfig = config?.analytics?.awsPlugin;
+    if (analyticsConfig == null) {
+      throw const AnalyticsException('No Pinpoint plugin config available');
+    }
+
+    if (!_isConfigured) {
+      // Initialize listeners
+      onTokenReceived.listen(_tokenReceivedListener);
+      onNotificationReceivedInForeground
+          .listen(_foregroundNotificationListener);
+      onNotificationOpened.listen(_notificationOpenedListener);
+      // Initialize Endpoint Client
+      await _serviceProviderClient.init(
+        config: config,
+        authProviderRepo: authProviderRepo,
+      );
+    }
+
+    // TODO: Register the callback dispatcher
+
+    _logger.info('CONFIGURE API | Successfully configure push notifications');
+
+    _isConfigured = true;
+  }
+
+  void _foregroundNotificationListener(
+    PushNotificationMessage pushNotificationMessage,
+  ) {
+    _logger.info(
+      'Successfully listrening to foreground events: $pushNotificationMessage',
+    );
+
+    // TODO(Samaritan1011001): Record Analytics
+  }
+
+  void _backgroundNotificationListener(
+    PushNotificationMessage pushNotificationMessage,
+  ) {
+    // TODO(Samaritan1011001): Record Analytics
+  }
+
+  void _notificationOpenedListener(
+    PushNotificationMessage pushNotificationMessage,
+  ) {
+    _logger.info(
+      'Successfully listrening to notification opened events: $pushNotificationMessage',
+    );
+    // TODO(Samaritan1011001): Record Analytics
+  }
+
+  String _tokenReceivedListener(
+    String address,
+  ) {
+    _logger.info('Successfully fetched the address: $address');
+    _registerDevice(address);
+    return address;
+  }
+
+  Future<void> _registerDevice(String address) async {
```

Comment:
Block `configure` from completing until this done and raise any exceptions encountered in `configure`

Reply:
Made the change 👍